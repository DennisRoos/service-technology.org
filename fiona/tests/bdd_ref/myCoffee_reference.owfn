{ coffee }

PLACE
INTERNAL
  p0,  { initial }
  p1,		
  p2,
  p3,
  p4;
INPUT
  euro, 
  tea,
  coffee;
OUTPUT
  beverage,
  money;

INITIALMARKING
  p0:	1
 ;

FINALMARKING
  p4:	1
 ;

TRANSITION t1	 { ?euro }
CONSUME
  p0:	1,
  euro: 1;	 
PRODUCE
  p1:	1;

TRANSITION t2	 { ?coffee }
CONSUME
  coffee:	1,
  p1: 1;
PRODUCE
  p2:	1;

TRANSITION t3	 { ?tea }
CONSUME
  p1:	1,
  tea : 1;
PRODUCE
  p2:	1;

TRANSITION t4	 { !beverage }
CONSUME
  p2:	1;
PRODUCE
  beverage:1,
  p3 : 1;
 
TRANSITION t5	 { !money }
CONSUME
  p3:	1;
PRODUCE
  money:1,
  p4 : 1;
  
TRANSITION t6	 { !money }
CONSUME
  p2:	1;
PRODUCE
  money:1,
  p4 : 1;  

{ END OF FILE }
