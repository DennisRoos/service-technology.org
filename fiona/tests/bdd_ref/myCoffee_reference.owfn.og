INTERFACE
  INPUT
    beverage,
    money;
  OUTPUT
    coffee,
    euro,
    tea;

NODES
  0 : (!coffee + !euro + !tea) : blue,
  1 : !euro : blue,
  6 : ?money : blue,
  7 : (?beverage * final) : blue : finalnode,
  15 : final : blue : finalnode,
  24 : ?money : blue,
  32 : !euro : blue,
  33 : ?money : blue,
  34 : (!coffee + !tea) : blue;

INITIALNODE
  0;

TRANSITIONS
  0 -> 1 : !coffee,
  0 -> 32 : !tea,
  0 -> 34 : !euro,
  1 -> 6 : !euro,
  6 -> 7 : ?money,
  6 -> 24 : ?beverage,
  7 -> 15 : ?beverage,
  24 -> 15 : ?money,
  32 -> 33 : !euro,
  33 -> 7 : ?money,
  33 -> 24 : ?beverage,
  34 -> 6 : !coffee,
  34 -> 33 : !tea;
