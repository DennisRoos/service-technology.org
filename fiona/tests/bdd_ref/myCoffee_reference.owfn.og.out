digraph g1 {
graph [fontname="Helvetica", label="OG of tests/bdd_ref/myCoffee_reference.owfn (parameters: -m1 -E)"];
node [fontname="Helvetica" fontsize=10];
edge [fontname="Helvetica" fontsize=10];
p0 [label="# 0\n(!coffee + !euro + !tea)", fontcolor=black, color=blue];
p0->p1 [label="!coffee", fontcolor=black, color=blue];
p1 [label="# 1\n!euro", fontcolor=black, color=blue];
p1->p6 [label="!euro", fontcolor=black, color=blue];
p6 [label="# 6\n?money", fontcolor=black, color=blue];
p6->p7 [label="?money", fontcolor=black, color=blue];
p7 [label="# 7\n(?beverage * final)", fontcolor=black, color=blue, peripheries=2];
p7->p15 [label="?beverage", fontcolor=black, color=blue];
p15 [label="# 15\nfinal", fontcolor=black, color=blue, peripheries=2];
p6->p24 [label="?beverage", fontcolor=black, color=blue];
p24 [label="# 24\n?money", fontcolor=black, color=blue];
p24->p15 [label="?money", fontcolor=black, color=blue];
p0->p32 [label="!tea", fontcolor=black, color=blue];
p32 [label="# 32\n!euro", fontcolor=black, color=blue];
p32->p33 [label="!euro", fontcolor=black, color=blue];
p33 [label="# 33\n?money", fontcolor=black, color=blue];
p33->p7 [label="?money", fontcolor=black, color=blue];
p33->p24 [label="?beverage", fontcolor=black, color=blue];
p0->p34 [label="!euro", fontcolor=black, color=blue];
p34 [label="# 34\n(!coffee + !tea)", fontcolor=black, color=blue];
p34->p6 [label="!coffee", fontcolor=black, color=blue];
p34->p33 [label="!tea", fontcolor=black, color=blue];
}